backend/src/agent/persistence/worker/workflow_saver.go

```diff
@@ -53,7 +56,12 @@ func (s *WorkflowSaver) Save(key string, namespace string, name string, nowEpoch
 			"Workflow (%s): transient failure: %v", key, err)
 
 	}
-
+	if wf.PersistedFinalState() && time.Now().Unix()-wf.FinishedAt() < s.ttlSecondsAfterWorkflowFinish {
```

Comment:
Have you considered to delete the workflow and add the persistent label in PA? It's confusing to have the logic in two different places.

Reply:
the persistence agent for now is read only and i think it's better to continue keep it that way. 
It will keep the roles and responsibility clean. WDYT?